arch/arc/core/atomic.S

```diff
@@ -417,6 +420,6 @@ SECTION_FUNC(TEXT, atomic_cas)
 
 	/* failed comparison */
 nanoAtomicCas_fail:
-	scond r1, [r0]	/* write old value to clear the access lock */
+	scond r3, [r0]	/* write old value to clear the access lock */
```

Comment:
It seems that llock and scond must be used in pair.  that's why scond is added at the end.
Based on your experience on linux,  is it ok to use llocak only ?

Reply:
`llock`/`scond` indeed are meant to be used as a pair but in their essence they are just a little bit specific load (`ld` on ARC parlance) & store (`st`) instructions. In fact you may use `llock` instead of normal `ld` easily and it will do the job - it will read specified memory location on each invocation.

But it's `scond` ("store conditional") which makes that pair useful for implementation of atomic data operations because `scond` only succeeds if it sees hidden architectural flag set previously by `llock` and not yet reset by IRQ, exception or any write to that same location.

That said it's OK to not execute `scond` if there's no need to like in this case - we already know data loaded by `llock` doesn't match our expectations. Again there's no requirement to use `scond` if `llock` was used previously.

Reply:
OK, I will try and test it on HSDK, if no problem , i will optimize it as you suggested